lib/charms/opensearch/v0/opensearch_distro.py

```diff
@@ -430,16 +435,55 @@ def current(self) -> Node:
                 unit_number=self._charm.unit_id,
                 temperature=current_node.get("attributes", {}).get("temp"),
             )
+
         except OpenSearchHttpError:
-            # we try to get the most accurate description of the node
-            conf_on_disk = self.config.load("opensearch.yml")
+
+            # we try to get the most accurate description of the node from the static config
+            conf = self.config.load("opensearch.yml")
+
+            # also, if possible we rely on the Deployment Description (databag)
+            deployment_desc = self._charm.opensearch_peer_cm.deployment_desc()
+
+            # Application Priority: Deployment Description
+            # Reason: No reason to re-construct the App object
+            #  - it's available 99% of scenarios
+            #  - it's the same object as a re-constructed one (i.e. no dynamic changes on App)
+            if deployment_desc is None:
+                app = App(id=conf.get("node.attr.app_id"))
+            else:
+                app = deployment_desc.app
+
+            # Roles (Temperature) Priority: local config
+            # Reason:
+            #  - Deployment Description is holding "expected state" (that may not be applied)
+            #  - Static config holds the currently applied settings
+            try:
+                roles = conf["node.roles"]
+            except KeyError:
+                if deployment_desc and deployment_desc.start == StartMode.WITH_PROVIDED_ROLES:
+                    roles = deployment_desc.config.roles
+                else:
+                    roles = GeneratedRoles
+
+            if not roles:
+                raise OpenSearchError("Can not determine app name and/or roles.")
+
+            temperature = None
+            try:
+                temperature = conf["node.attr.temp"]
+            except KeyError:
+                if deployment_desc:
```

Comment:
I believe this is not needed - as we'd have been into `if not roles` and raised an exception.

Reply:
Redundant indeed. I'll remove it.

Reply:
Keeping this check after modifications above. Currenty it's not redundant.

It _is_ (technically) possible that we may have a config file, but there would no `node.attr.temp` (while it _did_ include `node.roles`, thus no exceptions above were triggered). 
Now it's accepted **not** to have a Deployment Description at this point. Critical data (`app`, `roles`) was retrieved from config, we're all good -- yet (optional/missable) temperature is missing from the config. See new unittests (clearly corner-cases).

Temperature value detection should _only_ fall back to Deployment Description in case the latter exists.